### ISSUE TITLE
[BUG] TiledKNNTest fails

### ISSUE THREAD

**Issue by enp1s0** [rapidsai/raft#1526]
There is an issue with the TiledKNNTest in the latest branch-23.06 (commit 8e412b4), as it fails. However, it works properly when reverting 6fdb041 .

@benfred, can you provide any insight into what might be causing this issue?

## test log
```
[ RUN      ] TiledKNNTest/TiledKNNTestF.BruteForce/15
/opt/conda/conda-bld/work/cpp/test/neighbors/tiled_knn.cu:174: Failure
Value of: raft::spatial::knn::devArrMatchKnnPair(ref_indices_.data(), raft_indices_.data(), ref_distances_.data(), raft_distances_.data(), num_queries, k_, float(0.001), stream_)
  Actual: false (actual=19258,4.8802375793457031!=expected19829,4.8662505149841309 @0,0)
Expected: true
```

ref: https://github.com/rapidsai/raft/actions/runs/5010580599/jobs/8981566095?pr=1514

**Comment by benfred** [comment]
@enp1s0 thanks for the heads up - I'm taking a look

**Comment by benfred** [comment]
It looks like the problem was just in the unittest - with the test expecting sorted outputs, and the matrix::select_k not guaranteeing that outputs will be sorted. Fix is in #1533

**Comment by enp1s0** [comment]
Thanks for fixing the bug!
